internal/datacoord/garbage_collector_test.go

```diff
@@ -51,6 +53,132 @@ import (
 	"github.com/milvus-io/milvus/internal/util/funcutil"
 )
 
+type GarbageCollectorSuite struct {
+	suite.Suite
+
+	mockChunkManager *mocks.ChunkManager
+	gc               *garbageCollector
+}
+
+func (s *GarbageCollectorSuite) SetupTest() {
+	meta, err := newMemoryMeta()
+	s.Require().NoError(err)
+	s.mockChunkManager = &mocks.ChunkManager{}
+	s.gc = newGarbageCollector(
+		meta, newMockHandler(), GcOption{
+			cli:              s.mockChunkManager,
+			enabled:          true,
+			checkInterval:    time.Millisecond * 10,
+			missingTolerance: time.Hour * 24,
+			dropTolerance:    time.Hour * 24,
+		},
+	)
+}
+
+func (s *GarbageCollectorSuite) TearDownTest() {
+	s.mockChunkManager = nil
+	s.gc.close()
+	s.gc = nil
+}
+
+func (s *GarbageCollectorSuite) TestBasicOperation() {
+	s.Run("normal_gc", func() {
+		gc := s.gc
+		s.mockChunkManager.EXPECT().RootPath().Return("files")
+		s.mockChunkManager.EXPECT().ListWithPrefix(mock.Anything, mock.AnythingOfType("string"), mock.AnythingOfType("bool")).
+			Return([]string{}, []time.Time{}, nil)
+		gc.start()
+		// make ticker run at least once
+		time.Sleep(time.Millisecond * 20)
+
+		s.NotPanics(func() {
+			gc.close()
+		})
+	})
+
+	s.Run("nil_client", func() {
+		// initial a new garbageCollector here
+		gc := newGarbageCollector(nil, newMockHandler(), GcOption{
+			cli:     nil,
+			enabled: true,
+		})
+
+		s.NotPanics(func() {
+			gc.start()
+		})
+
+		s.NotPanics(func() {
+			gc.close()
+		})
+	})
+}
+
+func (s *GarbageCollectorSuite) TestScan() {
+	s.Run("listCollectionPrefix_fails", func() {
+		s.mockChunkManager.ExpectedCalls = nil
+		s.mockChunkManager.EXPECT().RootPath().Return("files")
+		s.mockChunkManager.EXPECT().ListWithPrefix(mock.Anything, mock.AnythingOfType("string"), mock.AnythingOfType("bool")).
+			Return(nil, nil, errors.New("mocked"))
+
+		s.gc.scan()
+		s.mockChunkManager.AssertNotCalled(s.T(), "Remove", mock.Anything, mock.Anything)
+	})
+
+	s.Run("collectionPrefix_invalid", func() {
+		s.mockChunkManager.ExpectedCalls = nil
+		s.mockChunkManager.EXPECT().RootPath().Return("files")
+		s.mockChunkManager.EXPECT().ListWithPrefix(mock.Anything, mock.AnythingOfType("string"), mock.AnythingOfType("bool")).
+			Return([]string{"files/insert_log/1/", "files/bad_prefix", "files/insert_log/string/"}, lo.RepeatBy(3, func(_ int) time.Time {
+				return time.Now().Add(-time.Hour)
+			}), nil)
+		s.gc.option.collValidator = func(collID int64) bool {
```

Comment:
Need to mock files consist of both valid and invalid collections, and then check whether only the valid collections were removed

Reply:
fixed